### PR TITLE
games-strategy/fheroes2: new package, add 1.0.0

### PR DESCRIPTION
**Pull request by winterheart**
Closes: https://bugs.gentoo.org/279781
Signed-off-by: Azamat H. Hackimov <azamat.hackimov@gmail.com>

### CODE REVIEW

**Comment by dlan17** [comment]
hi @winterheart , as already pointed, proxy-maint project normally do not accept new packages, and GURU project is a better place to go, please check [1]

[1] https://wiki.gentoo.org/wiki/Project:GURU

File: games-strategy/fheroes2/metadata.xml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+<email>proxy-maint@gentoo.org</email>
+<name>Proxy Maintainers</name>
+</maintainer>
+<upstream>
+<remote-id type="github">ihhub/fheroes2</remote-id>
+</upstream>
+<use>
+<flag name="image">Enable sdl2-image support</flag>
+</use>
+</pkgmetadata>
```
Comment (juippis) [inline games-strategy/fheroes2/metadata.xml:18]: Weird indentation.

File: games-strategy/fheroes2/fheroes2-1.0.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="+image"
+
+BDEPEND="sys-devel/gettext"
+DEPEND="
+	image? ( media-libs/sdl2-image[png] )
+	media-libs/libsdl2
+	media-libs/sdl2-mixer
+	sys-libs/zlib:=
```
Comment (juippis) [inline games-strategy/fheroes2/fheroes2-1.0.0.ebuild:22]: Please sort unconditional deps before conditional ones.

Comment (juippis) [inline games-strategy/fheroes2/fheroes2-1.0.0.ebuild:22]:
```suggestion
	media-libs/libsdl2
	media-libs/sdl2-mixer
	sys-libs/zlib
	image? ( media-libs/sdl2-image[png] )
```

File: games-strategy/fheroes2/fheroes2-1.0.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+src_prepare() {
+	default
+	sed -i -e "s:doc/README.txt:doc/INSTALL.md:g" -e "s:LICENSE::g" \
+		CMakeLists.txt || die
+	cmake_src_prepare
+}
+
+src_configure() {
+	mycmakeargs=(
+		-DUSE_SYSTEM_LIBSMACKER=OFF # broken
```
Comment (juippis) [inline games-strategy/fheroes2/fheroes2-1.0.0.ebuild:37]: Is there an upstream issue that can be linked to track this?

File: games-strategy/fheroes2/fheroes2-1.0.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+		-DENABLE_IMAGE=$(usex image)
+		-DENABLE_TOOLS=OFF
+	)
+	cmake_src_configure
+}
+
+pkg_postinst() {
+	elog "In order to play you need copy of HoMM II or HoMM II: The Price of"
+	elog "Loyalty data. Please read /usr/share/doc/${PF}/INSTALL.md for"
+	elog "futher instructions."
```
Comment (juippis) [inline games-strategy/fheroes2/fheroes2-1.0.0.ebuild:48]:
```suggestion
	elog "further instructions."
```

**Review by juippis — COMMENTED** [review]
@gentoo/games interested?

File: games-strategy/fheroes2/fheroes2-1.0.0.ebuild (hunk 3 above)
Comment (winterheart) [reply games-strategy/fheroes2/fheroes2-1.0.0.ebuild:37]: Buildsystem tries to search dev-games/libsmacker via pkg-config, but system package don't provides .pc-files. Actually they uses customized bundled libsmacker, and they stated that vanilla libsmacker broken to play actual data.

**Pull request closed without merging**